Automattic/sensei
# Fix end to e2e  fails caused by nonce issue

Resolves #6570


## Proposed Changes
* Use a separate browser context to run the API calls. 
* Create a new admin user to only run the API calls without messing the test that requires admin access.
* Implement a retry system log in the API user if there is a fail to get the nonce. 

We previously were using the same request context to make API calls as admin and run the e2e test as the relevant user, it was messing the context with different cookies, as explained [here](https://playwright.dev/docs/api/class-apirequestcontext#:~:text=If%20you%20want%20API%20requests%20to%20not%20interfere%20with%20the%20browser%20cookies%20you%20should%20create%20a%20new%20APIRequestContext%20by%20calling%20apiRequest.newContext().%20Such%20APIRequestContext%20object%20will%20have%20its%20own%20isolated%20cookie%20storage.) 


## Testing Instructions
* Go to this branch and introduce testing changes and check if the pipeline is failing with the error `apiRequestContext.get: 400 Bad Request`



## Pre-Merge Checklist
